### PR TITLE
Adds a way to disable jobs based on the gamemode(and disable security on gang mode with that)

### PR DESCRIPTION
**Pull request by lzimann**
Now you can disable certain jobs in a gamemode.
It is a config option.

Gang now has no security when that config is enabled(it also deletes the armory in round start).

### CODE REVIEW

**Comment by GunHog** [comment]
I think you should split these PRs. The job disabling feature on its own is fine, but not everyone is going to agree with your gang balance changes, and your feature is likely to get held up because of it.

**Comment by lzimann** [comment]
i don't really care if it gets held up

**Comment by PKPenguin321** [comment]
Being able to meta that it's gang is kinda lame but this is still super cool

**Comment by lzimann** [comment]
it takes like 2 minutes to know it is gang with or without this anyway

**Comment by KorPhaeron** [comment]
cool

**Comment by InsaneHyena** [comment]
We have to at least test merge this. Gang mode is shit and beyond redemption, but maybe it can at least improve it a little.

**Comment by TechnoAlchemisto** [comment]
if this is what your solution is I recommend lowering the weight of gang

not that it's bad, it's just that it's actually a pretty big change to just not have security at all

**Comment by lzimann** [comment]
honestly i prefer just disabling gang because fuck this gamemode to be honest

**Comment by TechnoAlchemisto** [comment]
I feel 100% the same

**Comment by Incoming5643** [comment]
>Gang now has no security when that config is enabled(it also deletes the armory in round start).

I have literally never see a tell as telly as this. I'm not speaking on the merits of the idea here but goddamn.

**Comment by KorPhaeron** [comment]
You already know its gang as soon as you see two or more tags of the same colour

**Comment by Bawhoppen** [comment]
that gang part is dumb

**Comment by Bawhoppen** [comment]
to elaborate, I liked the asymmetrical design of security vs the gangs

**Comment by Bawhoppen** [comment]
to elaborate, I liked the asymmetrical design of security vs the gangs

**Comment by Bawhoppen** [comment]
to elaborate, I liked the asymmetrical design of security vs the gangs

**Comment by Bawhoppen** [comment]
to elaborate, I liked the asymmetrical design of security vs the gangs

**Comment by TechnoAlchemisto** [comment]
I'm getting a feeling you liked the asymmetrical design of security vs the gangs

**Comment by Bawhoppen** [comment]
on my side it didn't look like it posted

File: code/controllers/configuration.dm [hunk 1]
```diff
@@ -59,6 +59,7 @@
 	var/jobs_have_minimal_access = 0	//determines whether jobs use minimal access or expanded access.
 	var/jobs_have_maint_access = 0 		//Who gets maint access?  See defines above.
 	var/sec_start_brig = 0				//makes sec start in brig or dept sec posts
+	var/gamemode_disable_jobs = 0 		//Disables job if the gamemode supports
```
Comment (optimumtact) [inline code/controllers/configuration.dm]: why not use the disabled jobs list instead? if it's null it doesn't disable jobs, if it's got items, it's enabled.

File: code/game/gamemodes/gang/gang.dm [hunk 2]
```diff
@@ -66,6 +70,13 @@ var/list/gang_colors_pool = list("red","orange","yellow","green","blue","purple"
 		boss.restricted_roles = restricted_jobs
 		log_game("[boss.key] has been selected as the Boss for the [G.name] Gang")
 
+	if(config.gamemode_disable_jobs) // It deletes all guns from armory
```
Comment (optimumtact) [inline code/game/gamemodes/gang/gang.dm]: this isn't even related to disabling jobs?

File: code/game/gamemodes/gang/gang.dm [hunk 3]
```diff
@@ -23,12 +23,15 @@ var/list/gang_colors_pool = list("red","orange","yellow","green","blue","purple"
...
 	config_tag = "gang"
 	antag_flag = ROLE_GANG
+	disabled_jobs = list("Security Officer", "Warden", "Detective", "Head of Security")
 	restricted_jobs = list("Security Officer", "Warden", "Detective", "AI", "Cyborg","Captain", "Head of Personnel", "Head of Security", "Chief Engineer", "Research Director", "Chief Medical Officer")
 	required_players = 20
 	required_enemies = 2
 	recommended_enemies = 2
 	enemy_minimum_age = 14
 
+	var/disable_security = FALSE
```
Comment (optimumtact) [inline code/game/gamemodes/gang/gang.dm]: what is the point of this var?

**Comment by optimumtact** [comment]
The idea is interesting, but the change to gang mode just papers over the cracks in that mode, it should just be removed.

**Comment by lzimann** [comment]
eh i'm way too lazy to do this now, feel free to make a pr if anyone cares enough about this

**Comment by KorPhaeron** [comment]
boo

**Comment by PKPenguin321** [comment]
Somebody please salvage this, we need it

**Comment by lzimann** [comment]
it's a simple code jsut copy from my pr

**Comment by optimumtact** [comment]
add their repo as a remote and pull the branch, a far superior method of retrieving their changes.

**Pull request closed without merging**